[browser] profiler setup and env variables

pavelsavara opened the pull request:
- set `DOTNET_DiagnosticPorts` by `DiagnosticPorts` MSBuild property, same as other mono platforms
- new `WasmEnvironmentVariable` item list
- new `WasmAppBuilder.EnvVariables`
- set `FeaturePerfTracing` when `WasmPerfTracing`
- fixed `enablePerfTracing` || `enableBrowserProfiler` for profiler events `cwraps`
- simplified mono profilers setup
- split `LogProfilerTest` and `BrowserProfilerTest`

Contributes to https://github.com/dotnet/runtime/issues/76316

pavelsavara committed:
- set FeaturePerfTracing when WasmPerfTracing

- new WasmEnvironmentVariable item list
- new WasmAppBuilder.EnvVariables
- fixed enablePerfTracing || enableBrowserProfiler for profiler events
- simplified mono profiler setup
- split LogProfilerTest and BrowserProfilerTest

pavelsavara committed:
Merge branch 'main' into browser_env

pavelsavara committed:
fix

javiercn reviewed (approved):
Changes look good to me, nothing strikes me as incorrect, although I don't have a lot of context.

My understanding is that this is enabled through the build, and then flows down to the runtime, with the most relevant change being the ability to add environment variables through the build on demand (which was not possible before).

Am I missing something?

ilonatommy commented on src/mono/browser/runtime/cwraps.ts:
```diff
@@ -130,9 +130,9 @@ const fn_signatures: SigLine[] = [
-    [() => !runtimeHelpers.emscriptenBuildOptions.enablePerfTracing, "mono_jiterp_prof_enter", "void", ["number", "number"]],
-    [() => !runtimeHelpers.emscriptenBuildOptions.enablePerfTracing, "mono_jiterp_prof_samplepoint", "void", ["number", "number"]],
-    [() => !runtimeHelpers.emscriptenBuildOptions.enablePerfTracing, "mono_jiterp_prof_leave", "void", ["number", "number"]],
+    [() => !runtimeHelpers.emscriptenBuildOptions.enablePerfTracing || runtimeHelpers.emscriptenBuildOptions.enableBrowserProfiler, "mono_jiterp_prof_enter", "void", ["number", "number"]],
```
In these 3 lines we are negating `enablePerfTracing` to register jiterpreter's methods (register if profiler is off) but then, in src/mono/browser/runtime/jiterpreter.ts we don't use that negation. It's not this PR's change but I don't follow

ilonatommy replied:
It's lazy loading, not skipping the method. So we are lazy loading if `enablePerfTracing` is off but `enableBrowserProfiler` on? Why?

pavelsavara replied:
this is typo, many thanks!

pavelsavara commented:
> (which was not possible before).

It was/is possible on runtime, there is JS and managed API for it. 
It was also possible to hack it via specially crafted value of `WasmExtraConfig` msbuild items. But it's very clumsy.

pavelsavara committed:
feedback

pavelsavara committed:
feedback from @maraf

pavelsavara committed:
Merge branch 'browser_env' of https://github.com/pavelsavara/runtime into browser_env

pavelsavara committed:
fix

pavelsavara committed:
fix

pavelsavara committed:
fix linker

- remove WASM_PERFTRACING from rollup
- remove mono_wasm_diagnostic_imports and include it always

pavelsavara committed:
fix minimal sample

pavelsavara committed:
Merge branch 'main' into browser_env
